src/bin/cli/stats/archive.rs

```diff
@@ -0,0 +1,180 @@
+use super::SnapshotCommandType;
+use catalyst_toolbox::stats::archive::load_from_csv;
+use catalyst_toolbox::stats::archive::load_from_folder;
+use catalyst_toolbox::stats::archive::ArchiveStats;
+use catalyst_toolbox::stats::block0::wallets::calculate_wallet_distribution_from_initials_utxo;
+use catalyst_toolbox::stats::snapshot::read_initials_from_file;
+use jormungandr_lib::interfaces::Initial;
+use std::collections::BTreeMap;
+use std::fmt::Debug;
+use std::path::{Path, PathBuf};
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+pub struct ArchiveCommand {
+    #[structopt(long = "csv", required_unless = "folder")]
+    pub csv: Option<PathBuf>,
+
+    #[structopt(long = "folder", required_unless = "csv")]
+    pub folder: Option<PathBuf>,
+
+    #[structopt(long = "output")]
+    pub output: Option<PathBuf>,
+
+    #[structopt(short = "d", long = "distribution")]
+    pub calculate_distribution: bool,
+
+    #[structopt(subcommand)]
+    pub command: Command,
+}
+
+impl ArchiveCommand {
+    pub fn exec(&self) -> Result<(), catalyst_toolbox::stats::Error> {
+        let archiver: ArchiveStats = {
+            if let Some(csv) = &self.csv {
+                load_from_csv(&csv)?.into()
+            } else if let Some(folder) = &self.folder {
+                load_from_folder(&folder)?.into()
+            } else {
+                panic!("no csv nor folder defined");
+            }
+        };
+
+        match &self.command {
+            Command::VotesByCaster => {
+                let result = archiver.number_of_votes_per_caster()?;
+                if self.calculate_distribution {
+                    self.write_to_file_or_print(ArchiveStats::calculate_distribution(&result)?)?;
+                } else {
+                    self.write_to_file_or_print(result)?;
+                }
+            }
+            Command::VotesBySlot => {
+                let result = archiver.number_of_tx_per_slot()?;
```

Comment:
Good point. there is a silent assumption that all transactions in archive are votes. Don't really know what should be the fix, renames `VotesBySlot` to `TxBySlot`, but then it's useless from client perspective because he doesn't know what is the subset of votes. On the other side catalyst archiver guarantees that those are votes. So maybe we should assume that all transactions in archiver are votesr and rename `number_of_tx_per_slot` to `number_of_votes_per_slot`